crates/common/tedge_config/src/settings.rs

```diff
@@ -172,6 +172,19 @@ impl ConfigSetting for MqttPortSetting {
     type Value = Port;
 }
 
+pub struct MqttBindAddressSetting;
+
+impl ConfigSetting for MqttBindAddressSetting {
+    const KEY: &'static str = "mqtt.bind_address";
+
+    const DESCRIPTION: &'static str = concat!(
+        "Mqtt bind address, which is used by the mqtt clients to publish or subscribe. ",
+        "Example: 0.0.0.0"
+    );
+
+    type Value = String;
```

Comment:
One thing I'm not a fun of in the config in general is that pretty much everything is a `String`. 

I would suggest to use a socket address type (eg `std::net::Ipv4Addr` or something different if hostnames should be allowed) like we did for some other configs. (I suppose this will have to allow for a domain name as well)

Reply:
Thought about it, kept it as a String to align with the other configurable IP addresses. For example external bind address. Maybe we can take it up as a separate PR to replace all of these with IPV4 and IPV6.

Reply:
Why not doing this in this PR? Both on the new property `MqttBindAddressSetting` and the previous one `MqttExternalBindAddressSetting`.

Reply:
Sure, I will implement in this PR itself.

Reply:
Yes, I am using the same